src/_pytest/python.py

```diff
@@ -741,6 +741,12 @@ def newinstance(self):
         return self.obj()
 
     def collect(self) -> Iterable[nodes.Item | nodes.Collector]:
+        if self.config.getini("discover_imports") == ("false" or False):
```

Comment:
We should ensure this is always a boolean or parsed somewhere else to ensure Separation

Reply:
I think changing the default value for the option to `False` and changing this to `if self.config.getinit("discover_imports"):` should be enough.